core/src/main/java/com/alibaba/nacos/core/distributed/raft/JRaftProtocol.java

```diff
@@ -123,36 +124,39 @@ public void init(RaftConfig config) {
-                    final List<String> raftClusterInfo = event.getRaftClusterInfo();
+            try {
+                NotifyCenter.registerSubscriber(new Subscriber<RaftEvent>() {
+                    @Override
+                    public void onEvent(RaftEvent event) {
+                        Loggers.RAFT.info("This Raft event changes : {}", event);
+                        final String groupId = event.getGroupId();
+                        Map<String, Map<String, Object>> value = new HashMap<>();
+                        Map<String, Object> properties = new HashMap<>();
+                        final String leader = event.getLeader();
+                        final Long term = event.getTerm();
+                        final List<String> raftClusterInfo = event.getRaftClusterInfo();
+                        
+                        // Leader information needs to be selectively updated. If it is valid data,
+                        // the information in the protocol metadata is updated.
+                        MapUtils.putIfValNoEmpty(properties, MetadataKey.LEADER_META_DATA, leader);
+                        MapUtils.putIfValNoNull(properties, MetadataKey.TERM_META_DATA, term);
+                        MapUtils.putIfValNoEmpty(properties, MetadataKey.RAFT_GROUP_MEMBER, raftClusterInfo);
+                        
+                        value.put(groupId, properties);
+                        metaData.load(value);
+                        
+                        // The metadata information is injected into the metadata information of the node
+                        injectProtocolMetaData(metaData);
+                    }
                     
-                    // Leader information needs to be selectively updated. If it is valid data,
-                    // the information in the protocol metadata is updated.
-                    MapUtils.putIfValNoEmpty(properties, MetadataKey.LEADER_META_DATA, leader);
-                    MapUtils.putIfValNoNull(properties, MetadataKey.TERM_META_DATA, term);
-                    MapUtils.putIfValNoEmpty(properties, MetadataKey.RAFT_GROUP_MEMBER, raftClusterInfo);
-                    
-                    value.put(groupId, properties);
-                    metaData.load(value);
-                    
-                    // The metadata information is injected into the metadata information of the node
-                    injectProtocolMetaData(metaData);
-                }
-                
-                @Override
-                public Class<? extends Event> subscribeType() {
-                    return RaftEvent.class;
-                }
-                
-            });
+                    @Override
+                    public Class<? extends Event> subscribeType() {
+                        return RaftEvent.class;
+                    }
+                });
+            } catch (NacosException ex) {
+                ex.printStackTrace();
```

Comment:
Use log printing

Reply:
And registered listeners should have no abnormal perception，Readjust the interface inside the NotifyCenter and adjust the BiFunction so that no exceptions are thrown

Reply:
I fix it.